packages/anoncreds/src/formats/AnonCredsCredentialFormatService.ts

```diff
@@ -303,10 +311,67 @@ export class AnonCredsCredentialFormatService implements CredentialFormatService
     const credentialRequest = requestAttachment.getDataAsJson<AnonCredsCredentialRequest>()
     if (!credentialRequest) throw new AriesFrameworkError('Missing anoncreds credential request in createCredential')
 
+    // We check locally for credential definition info. If it supports revocation, we need to search locally for
+    // an active revocation registry
+    const credentialDefinition = (
+      await agentContext.dependencyManager
+        .resolve(AnonCredsCredentialDefinitionRepository)
+        .getByCredentialDefinitionId(agentContext, credentialRequest.cred_def_id)
+    ).credentialDefinition.value
+
+    let revocationRegistryDefinitionId
+    let revocationStatusList
+    let tailsFilePath
+
+    if (credentialDefinition.revocation) {
+      const [revocationRegistryDefinitionPrivateRecord] = await agentContext.dependencyManager
+        .resolve(AnonCredsRevocationRegistryDefinitionPrivateRepository)
+        .findAllByCredentialDefinitionIdAndState(
+          agentContext,
+          credentialRequest.cred_def_id,
+          RevocationRegistryState.Active
+        )
+
+      // TODO: should a new revocation registry be generated/published/etc automatically?
+      if (!revocationRegistryDefinitionPrivateRecord) {
+        throw new AriesFrameworkError(`No available revocation registry found for ${credentialRequest.cred_def_id}`)
+      }
+
+      revocationRegistryDefinitionId = revocationRegistryDefinitionPrivateRecord.revocationRegistryDefinitionId
+
+      // get current revocation status list
+      const registryService = agentContext.dependencyManager.resolve(AnonCredsRegistryService)
+      const registry = registryService.getRegistryForIdentifier(agentContext, credentialRequest.cred_def_id)
+      const result = await registry.getRevocationStatusList(
```

Comment:
Very good point. It seems that revocation status list is required by `anoncreds-rs` to determine [if the index uses issuance by default or not](https://github.com/hyperledger/anoncreds-rs/blob/82e50f5e8215c4dbf4bb60ba65533716f5e1ae08/src/services/issuer.rs#L741).

If we are only supporting issuance by default, of course this processing does not make sense. I feel this should be addressed in anoncreds-rs, but as a workaround for this, might we send to the library a dummy revocation status list in this case?

Reply:
Workaround works for me! Maybe you can open an issue so we can track it

Reply:
Unfortunately, anoncreds-rs does an strict check of revocation status list object, so the 'dummy' object is not as dummy as we would like (it's needed to generate it through the library). This generation seems to be fast but I'm not completely sure how it will perform in large use cases. 

I opened https://github.com/hyperledger/anoncreds-rs/issues/227 to discuss this and confirm that our reasoning is correct. If we are right, it would be a nice update to the library to simplify both its API and processing time.